src/platforms/eglstream-kms/server/display.cpp

```diff
@@ -198,21 +195,37 @@ class DisplayBuffer
 
     void post() override
     {
+        using namespace std::chrono_literals;  // For operator""ms()
+
         // Wait for the last flip to finish, if it hasn't already.
         pending_flip.get();
-
         pending_flip = event_handler->expect_flip_event(
-            crtc_id,
+            output->crtc_id(),
             [this](unsigned frame_count, std::chrono::milliseconds frame_time)
             {
                 // TODO: Um, why does NVIDIA always call this with 0, 0ms?
                 display_report->report_vsync(
-                    crtc_id,
+                    output->crtc_id(),
                     mg::Frame {
                         frame_count,
                         mir::time::PosixTimestamp(CLOCK_MONOTONIC, frame_time)});
             });
 
+        if (next_swap) // This is the KMS route
+        {
+            auto const predicted_render_time = 50ms; // Predicted worst case render time for the next frame...
+
+            output->set_crtc(*next_swap);
```

Comment:
This will do an immediate, tearing CRTC update (ie: it will not wait for vsync). I don't think this will trigger the flip event handler, either?